# Multiple Linear Regression <br>
- Bir tane bağımlı değişken ile bununla ilişkisi olan bir dizi bağımsız değişken arasındaki ilişkiyi ortaya koymak için yapılan analizdir.

In [24]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression

simple linear regression = <strong>y = b0 + b1 * x</strong> <br>
yani,                maas = b0 + b1 * deneyim <br>
multiple LR = <strong>y = b0 + b1 * x1 + b2 * x2</strong> <br>
yani, maas = b0 + b1 * deneyim + b2 * yas 

maas = <strong> dependent variable</strong> (bağımlı) yani benim maaşım deneyim ve yaşa bağlı <br>
deneyim , yas= <strong> independet variable </strong> (bağımsız değişken) <br>
amaç = MİN(MSE) yani öyle bir b0 b1 b2 değeri seçeceğim ki MSE düşük olacak.

In [9]:
df=pd.read_csv("/Users/bestecetin/Desktop/machine_learning/courses/machinelearning/Regression/MultipleLinearRegression/data/multiple_linear_regression_dataset.csv",sep=";")

In [38]:
x = df.iloc[:,[0,2]].values #deneyim , yas sutununu aldık
y = df.maas.values.reshape(-1,1) #maas

In [39]:
x

array([[ 0.5, 22. ],
       [ 0. , 21. ],
       [ 1. , 23. ],
       [ 5. , 25. ],
       [ 8. , 28. ],
       [ 4. , 23. ],
       [15. , 35. ],
       [ 7. , 29. ],
       [ 3. , 22. ],
       [ 2. , 23. ],
       [12. , 32. ],
       [10. , 30. ],
       [14. , 34. ],
       [ 6. , 27. ]])

In [29]:
multiple_linear_regression = LinearRegression() 
multiple_linear_regression.fit(x,y)

LinearRegression()

In [30]:
print("b0: ", multiple_linear_regression.intercept_)

b0:  [10376.62747228]


In [31]:
print("b1,b2: ", multiple_linear_regression.coef_)

b1,b2:  [[1525.50072054 -416.72218625]]


In [32]:
multiple_linear_regression.predict(np.array([[10,35],[5,35]])) 

array([[11046.35815877],
       [ 3418.85455609]])

- 10 yıllık deneyim ve 35 yas  maas olarak = 11046
- 5 yıllık deneyim ve 35 yas maas olarak = 3418